src/site_model.cpp

```diff
@@ -34,16 +34,29 @@ void WeibullSiteModel::SetParameters(const EigenVectorXdRef param_vector) {
 // Equivalent to the discretized gamma method in Yang 1994.
 // The scale (lambda) is fixed to 1
 void WeibullSiteModel::UpdateRates() {
-  double mean = 0;
+  double mean_rate = 0;
+  double mean_rate_derivative = 0;
+  std::vector<double> deriv_unscaled_rates(category_count_);
   for (size_t i = 0; i < category_count_; i++) {
     double quantile = (2.0 * i + 1.0) / (2.0 * category_count_);
     // Set rate to inverse CDF at quantile.
     category_rates_[i] = pow(-std::log(1.0 - quantile), 1.0 / shape_);
-    mean += category_rates_[i];
+    mean_rate += category_rates_[i];
+    // Derivative of unormalized rate i wrt shape
```

Comment:
This is awesome! Do you have any helpful links for understanding the derivation?

Reply:
It is coming from physher and there is even less comments in the code